### PR TITLE
Adding OnePhotonSeries class

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
## Motivation

Adding the `OnePhotonSeries` class to PyNWB, which was recently added to the schema in https://github.com/NeurodataWithoutBorders/nwb-schema/pull/523

### CODE REVIEW

**Comment by CodyCBakerPhD** [comment]
@rly Looks like when I updated the nwb-schema version, it triggered some failing container-type tests? Any ideas?

**Comment by CodyCBakerPhD** [comment]
@rly @oruebel @bendichter OK with #1540 in this now appears to be working, ready for review

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## Upcoming
 
 ### Enhancements and minor changes
+- Added a class and tests for the `OnePhotonSeries`. @CodyCBakerPhD [#1593](https://github.com/NeurodataWithoutBorders/pynwb/pull/1593)
```
Comment (oruebel) [inline CHANGELOG.md]:
```suggestion
- Added a class and tests for the `OnePhotonSeries` new in NWB v2.6.0. @CodyCBakerPhD [#1593](https://github.com/NeurodataWithoutBorders/pynwb/pull/1593)(see also NWB Schema [#523](https://github.com/NeurodataWithoutBorders/nwb-schema/pull/523)
```

**Review by oruebel — CHANGES_REQUESTED** [review]
This looks in general good to me. Could you also add:
* A ``mock_OnePhotonSeries`` function to the mock test module ``src/pynwb/testing/mock/ophys.py`` Here the example for TwoPhotonSeries:
https://github.com/NeurodataWithoutBorders/pynwb/blob/5d8efdb43b0855556bcdd229a27a7bfaf1daf8fc/src/pynwb/testing/mock/ophys.py#L66
* A roundtrip test for One`PhotonSeries to ``tests/integration/hdf5/test_ophys.py``. Here the test for TwoPhotonSeries:
https://github.com/NeurodataWithoutBorders/pynwb/blob/5d8efdb43b0855556bcdd229a27a7bfaf1daf8fc/tests/integration/hdf5/test_ophys.py#L138

**Comment by CodyCBakerPhD** [comment]
@oruebel Adding the round-trip IO integration test indeed reveals an issue with one of the new fields - the relevant piece of the traceback is

```
 hdmf.build.errors.ConstructError: ..... 

Could not construct OnePhotonSeries object due to: OnePhotonSeries.__init__: incorrect type for 'binning' (
    got 'uint64', expected 'int'
)
```

any suggestions for how to fix this?

**Comment by CodyCBakerPhD** [comment]
In particular, the schema specification for this field is actually as `uint8`: https://github.com/NeurodataWithoutBorders/nwb-schema/blob/dev/core/nwb.ophys.yaml#L21

File: src/pynwb/ophys.py [hunk 2]
```diff
@@ -120,6 +120,82 @@ def __init__(self, **kwargs):
...
+        },
+        {
+            "name": "exposure_time",
+            "type": float,
+            "doc": "Exposure time of the sample; often the inverse of the frequency.",
+            "default": None,
+        },
+        {
+            "name": "binning",
+            "type": int,
```
Comment (rly) [inline src/pynwb/ophys.py]:
```suggestion
            "type": (int, "uint")
```

File: src/pynwb/ophys.py [hunk 3]
```diff
@@ -120,6 +120,82 @@ def __init__(self, **kwargs):
...
+        )
+    )
+    def __init__(self, **kwargs):
+        keys_to_set = (
+            "imaging_plane", "pmt_gain", "scan_line_rate", "exposure_time", "binning", "power", "intensity"
+        )
+        args_to_set = popargs_to_dict(keys_to_set, kwargs)
+        super().__init__(**kwargs)
+
+        for key, val in args_to_set.items():
```
Comment (rly) [inline src/pynwb/ophys.py:200]:
```suggestion

        if args_to_set["binning"] < 0:
            raise ValueError(f"Binning value must be >= 0: {args_to_set['binning']}")
        if isinstance(args_to_set["binning"], int):
            args_to_set["binning"] = np.uint(args_to_set["binning"])

        for key, val in args_to_set.items():
```

Comment (CodyCBakerPhD) [reply src/pynwb/ophys.py:200]: Indeed that appears to fix it - thanks!

Comment (CodyCBakerPhD) [reply src/pynwb/ophys.py:200]: (at least locally, let's see what CI says...)

File: src/pynwb/ophys.py [hunk 4]
```diff
@@ -120,6 +120,87 @@ def __init__(self, **kwargs):
...
+    )
+    def __init__(self, **kwargs):
+        keys_to_set = (
+            "imaging_plane", "pmt_gain", "scan_line_rate", "exposure_time", "binning", "power", "intensity"
+        )
+        args_to_set = popargs_to_dict(keys_to_set, kwargs)
+        super().__init__(**kwargs)
+
+        if args_to_set["binning"] < 0:
+            raise ValueError(f"Binning value must be >= 0: {args_to_set['binning']}")
```
Comment (oruebel) [inline src/pynwb/ophys.py:196]: Could you add a constructor test with a negative binning value to confirm that the ValueError is being raised correctly and ensure this line is covered

**Review by oruebel — DISMISSED** [review]

File: src/pynwb/ophys.py (hunk 4 above)
Comment (CodyCBakerPhD) [reply src/pynwb/ophys.py:196]: Added

**Review by oruebel — APPROVED** [review]

**Comment by oruebel** [comment]
WooHoo 100% test coverage on the diff. Thanks @CodyCBakerPhD for adding support for OnePhotonSeries!

@rly since this PR updates the schema, does this need to be merged with the dev branch or a different release branch?

**Comment by rly** [comment]
This should go to a new branch for schema 2.6 support. So that we try to stick with the release schedule, NWB 2.6 will be released around Jan. 5 and the corresponding PyNWB and MatNWB releases should be made around Jan. 8.

**Comment by rly** [comment]
I'll create that new branch and redirect this PR there.

**Comment by rly** [comment]
This is now good to merge @CodyCBakerPhD
